src/rules/newlineRule.ts

```diff
@@ -0,0 +1,150 @@
+/**
+* @license
+* Copyright 2017 Palantir Technologies, Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+ 
+import { getPreviousStatement, isBlock, isClassDeclaration, isFunctionWithBody, getChildOfKind } from "tsutils";
+import * as ts from "typescript";
+import * as Lint from "../index";
+ 
+enum OptionState {
+    always = "always",
+    ignore = "ignore",
+    never = "never",
+}
+ 
+const ALWAYS_IGNORE_OR_NEVER = {
+    enum: OptionState,
+    type: "string",
+};
+ 
+type OptionName = "return" | "class" | "functionBlock" | "block";
+const optionNames: OptionName[] = ["return", "class", "functionBlock", "block"];
+type Option = "always" | "ignore" | "never";
+type Options = Partial<Record<OptionName, Option>>;
+ 
+function parseOptions(json: Option | Options | undefined): Options {
+    const options: Options = { };
+    for (const optionName of optionNames) {
+        options[optionName] = typeof json === "object" ? json[optionName] : json === undefined ? "always" : json;
+    }
+    return options;
+}
```

Comment:
I'd suggest the following (note that this defaults to ignore, which might be more intuitive)
```ts
type Options = Record<OptionName, Option>;
function parseOptions(json: Partial<Options> | undefined) {
    const default: Options = {return: "ignore", class: "ignore", functionBlock: "ignore", block: "ignore"};
    return {...default, json};
}
```